formula-test/src/main/java/com/instacart/formula/test/TestActionObserver.kt

```diff
@@ -32,10 +49,20 @@ class TestActionObserver<Event>(
      * provide a [Cancelable].
      */
     fun cancel() {
-        val cancelable = cancelation ?: run {
+        assertNoErrors()
+
+        val cancelable = cancelable ?: run {
             throw IllegalStateException("Action did not return a cancelable.")
         }
 
         cancelable.cancel()
+
+        assertNoErrors()
+    }
+
+    private fun assertNoErrors() {
+        if (errors.isNotEmpty()) {
+            throw AssertionError("Expected no errors, but got: $errors")
+        }
```

Comment:
nit: use `Truth` assertions for better failure message and for better consistency across the project?

Reply:
Since it's part of the public API, I don't want to bring in an extra dependency within `formula-test` module.